src/package_test.ts

```diff
@@ -7,4 +7,14 @@ describe('package', () => {
         const v2 = require('../package-lock.json').version;
         expect(v1).to.equal(v2);
     });
+
+    it('package-lock should only reference official public npm registry', () => {
+        const deps = require('../package-lock.json').dependencies;
+        for (const key in deps) {
```

Comment:
It's not enough to loop through the top level - each of `dependencies` can have its own `dependencies`, e.g. https://github.com/kubernetes-client/javascript/blob/d18e69d531580db628a4e925f0ece42a80a20412/package-lock.json#L38

Reply:
done.